client/components/webcamModule.js

```diff
@@ -0,0 +1,114 @@
+angular.module('in-your-face')
+.controller('webcamModuleCtrl', function($scope) {
+  console.log("123");
+  var _video = null;
+  var patData = null;
+  var self = this;
+
+  this.patOpts = {x: 0, y: 0, w: 25, h: 25};
+
+  // Setup a channel to receive a video property
+  // with a reference to the video element
+  // See the HTML binding in main.html
+  this.channel = {};
+  //
+  // this.webcamError = false;
+  // this.onError = function (err) {
+  //     $scope.$apply(
+  //         function() {
+  //             self.webcamError = err;
+  //         }
+  //     );
+  // };
+  //
+  this.onSuccess = function () {
+          // The video element contains the captured camera data
+    _video = this.channel.video;
+    $scope.$apply(function() {
+      _video.width = 540;
+      _video.height =300;
+      self.patOpts.w = _video.width;
+      self.patOpts.h = _video.height;
+      this.showDemos = true;
+    });
+  };
+
+  var getVideoData = function getVideoData(x, y, w, h) {
+    var hiddenCanvas = document.createElement('canvas');
+    hiddenCanvas.width = _video.width;
+    hiddenCanvas.height = _video.height;
+    var ctx = hiddenCanvas.getContext('2d');
+    ctx.drawImage(_video, 0, 0, _video.width, _video.height);
+    return ctx.getImageData(x, y, w, h);
+  };
+
+    /**
+     * This function could be used to send the image data
+     * to a backend server that expects base64 encoded images.
+     *
+     * In this example, we simply store it in the scope for display.
+     */
+//
+  var sendSnapshotToServer = function sendSnapshotToServer(imgBase64) {
+    this.snapshotData = imgBase64;
+    console.log(this.snapshotData);
```

Comment:
Again, this log message will be base64 encoded data, wont be much use in log to see base64 encoded data. 
Next iteration we probably need a timestamp/image_id/hash to uniquely identify the image.